src/Language/PureScript/Errors.hs

```diff
@@ -500,33 +496,41 @@ prettyPrintSingleError full level e = prettyPrintErrorMessage . positionHintsFir
+      line $ "The type " ++ runProperName nm ++ " conflicts with a type class declaration of the same name."
     renderSimpleErrorMessage (CtorConflictsWithClass nm) =
-      line $ "Data constructor " ++ runProperName nm ++ " conflicts with type class declaration of the same name"
+      line $ "The data constructor " ++ runProperName nm ++ " conflicts with a type class declaration of the same name."
     renderSimpleErrorMessage (ClassConflictsWithType nm) =
-      line $ "Type class " ++ runProperName nm ++ " conflicts with type declaration of the same name"
+      line $ "The type class " ++ runProperName nm ++ " conflicts with a type declaration of the same name."
     renderSimpleErrorMessage (ClassConflictsWithCtor nm) =
-      line $ "Type class " ++ runProperName nm ++ " conflicts with data constructor declaration of the same name"
+      line $ "The type class " ++ runProperName nm ++ " conflicts with a data constructor declaration of the same name."
     renderSimpleErrorMessage (DuplicateModuleName mn) =
-      line $ "Module " ++ runModuleName mn ++ " has been defined multiple times."
+      line $ "The module " ++ runModuleName mn ++ " has been defined multiple times."
     renderSimpleErrorMessage (DuplicateClassExport nm) =
       line $ "Duplicate export declaration for type class " ++ runProperName nm
     renderSimpleErrorMessage (DuplicateValueExport nm) =
       line $ "Duplicate export declaration for value " ++ showIdent nm
     renderSimpleErrorMessage (CycleInDeclaration nm) =
-      line $ "Cycle in declaration of " ++ showIdent nm
+      paras [ line $ "The declaration of " ++ showIdent nm ++ " contains a cycle."
+            , line $ "The value of " ++ showIdent nm ++ " may not be defined here, so this reference is not allowed."
+            , line "You might be able to evaluate this expression lazily, using something like Control.Lazy.fix"
+            ]
     renderSimpleErrorMessage (CycleInModules mns) =
-      line $ "Cycle in module dependencies: " ++ intercalate ", " (map runModuleName mns)
-    renderSimpleErrorMessage (CycleInTypeSynonym pn) =
-      line $ "Cycle in type synonym" ++ foldMap ((" " ++) . runProperName) pn
+      line $ "There is a cycle in the module dependencies: " ++ intercalate ", " (map runModuleName mns)
+    renderSimpleErrorMessage (CycleInTypeSynonym name) =
+      paras [ line $ case name of
+                       Just pn -> "A cycle appears in the definition of the type synonym " ++ runProperName pn
+                       Nothing -> "A cycle appears in a set of type synonym definitions."
+            , line "Cycles are disallowed because they can lead to loops in the type checker."
+            , line "Consider using a 'newtype' instead."
+            ]
     renderSimpleErrorMessage (NameIsUndefined ident) =
-      line $ showIdent ident ++ " is undefined"
+      line $ "The name " ++ showIdent ident ++ " is undefined."
```

Comment:
"The value" again? I think everywhere we use "the name" when referring to values is probably better?